lib/attr_extras/attr_initialize/attributes.rb

```diff
@@ -0,0 +1,36 @@
+module AttrExtras
+  class AttrInitialize
+    class Attributes
+      def initialize(names_with_array_of_hash_arguments)
+        @names_with_array_of_hash_arguments = names_with_array_of_hash_arguments
+      end
+
+      def positional
+        @attributes_names ||= names_with_array_of_hash_arguments.take_while { |name_or_array| !name_or_array.is_a?(Array) }
+      end
```

Comment:
Yeah. Another option is to do nothing and assume people will use the lib sensibly, but I prefer not to trust people if I can help it ;D